src/main/java/com/fasterxml/jackson/databind/util/LookupCache.java

```diff
@@ -0,0 +1,29 @@
+package com.fasterxml.jackson.databind.util;
+
+import com.fasterxml.jackson.core.util.Snapshottable;
+
+import java.util.function.BiConsumer;
+
+/**
+ * An interface describing the required API for the Jackson-Databind Type cache.
+ *
+ * @see com.fasterxml.jackson.databind.type.TypeFactory#withCache(LookupCache<java.lang.Object,com.fasterxml.jackson.databind.JavaType>)
+ * @see SimpleLookupCache
+ */
+public interface LookupCache <K,V>
+        extends Snapshottable<LookupCache<K,V>>,
+        java.io.Serializable {
+    void contents(BiConsumer<K,V> consumer);
+    int size();
+
+    /**
+     * NOTE: key is of type Object only to retain binary backwards-compatibility
```

Comment:
Is that needed?

Reply:
I kept this from existing SimpleLookupCache documentation

Reply:
Ok. We wouldn't have to retain that (binary compatibility not preserved across 2.x -> 3.0 boundary) but... I don't feel strongly either way. I know that casting to type variable here won't be of much use as it's truly generic type. But it might help catch some type mismatches.

So I am fine either way at this point.